processor/metric/payload.go

```diff
@@ -37,12 +37,14 @@ var (
 	processorEntry  = common.MapStr{"name": processorName, "event": docType}
 )
 
-type sample interface {
-	transform(common.MapStr) error
+type sample struct {
+	name  string
+	value float64
+	unit  *string
 }
 
 type metric struct {
-	samples   []sample
+	samples   []*sample
```

Comment:
out of curiosity - why the change?

Reply:
changed sample from an interface to a struct since there is only one sample type now